provider/internal/java/util.go

```diff
@@ -127,10 +127,10 @@ func decompile(ctx context.Context, log logr.Logger, filter decompileFilter, wor
 					continue
 				}
 				cmd := exec.CommandContext(
-					ctx, "java", "-jar", "/bin/fernflower.jar", job.inputPath, outputPathDir)
+					ctx, "java", "-jar", "/usr/local/bin/fernflower.jar", job.inputPath, outputPathDir)
```

Comment:
This will require us to fix [the Dockerfile in https://github.com/konveyor/java-analyzer-bundle so that ](https://github.com/konveyor/java-analyzer-bundle/blob/9cb5a19d87531487df378afbb37de4c60d16ae87/Dockerfile#L35), is his change an abosolute requirement?

Reply:
Correct, but it would make decompilation more portable as pointed out in b52929fb86eb7d455e9acf4af97b13cd7de40f09. I could not test with this path on MacOS, as /bin is write protected (for good reasons).

In general I would appreciate if such paths were configurable by file or environment. We could do this in the long run. 

Having said that, my inner monk ruffles up at the thought of having a jar file in the /bin folder (even if it's only a container, and even if the file is executable), as the /bin folder traditionally is the core folder for OS binaries. User add-ons should go to /usr/local or /opt.

Reply:
I see that the main reason for this change is debugging. I'll send you some indications about how to debug from outside the container. I don't object to this change, but bare in mind that the analyzer will mostly be used from within the konveyor ecosystem in a containerized way, and that is the path we're mainly following.

Reply:
I fully understand that the end-user will mostly work with containers (which is perfectly fine for me) in order to avoid tons of local configurations, executables/tools etc. But to be honest, as a newbie development contributor, it is hard to follow all of these dependencies, and even harder to maintain them. 

Maybe for this change it is not much of a problem. But check out the change of Maven to 3.9 to avoid the compile goal (#393): 
* To get it running, 
  * you need to change the base image with the Java/Maven installations. 
  * Then build the analyzer-lsp image based on that. 
  * Finally build the Kantra image. 
* During runtime, 
  * `kantra` calls `konveyor-analyzer` (and some other tools), 
  * which call `mvn` at some point. 
If something goes wrong, it is hard to debug as the tools simply abort without leaving much trace to the root cause (perhaps I need to checkout the Jaeger integration?). I never saw that Maven failed (or even why) until I started the debugger.

To me it is much easier to run/debug all of this locally and have full control over each of the components.
Hence, I have the strong wish to be able to work locally with some (well-documented!) prerequisites as well as in containers. Setup should support both modes in my opinion.

BTW: Even when working completely in containers (as the end-user will do) it is essential to have a thorough documentation of all components, their workflows, all (intermediate) artifacts, and in particular how to observe their behaviour (logging/tracing/monitoring). Sorry, for repeating this over and over again ;-)

Reply:
While I agree in principle, the word support here is something to consider. 

Because the setup for the components is pretty complex, I worry about supporting everyone's local environments. This could be a long tail that would cost a lot of cycles, IMO. 

I would like to know if we can get an agreement on documentation and best effort to answer questions when testing locally, but there are no guarantees on time/fixes vs. in the container where there should be guarantees on this. Thoughts?

Reply:
Oh, and in addition, I am plus on on moving things out of `/bin/` that seems like a good citizen thing to do.

Reply:
Starting a smooth migration to better local executability / improving abilities to debug does not necessarily mean, that all possible paths must be fully supported. In this case, the change is invisible to almost everyone but makes life easier for someone developing outside of containers while not harming anyone else.